widgetry/src/widgets/button.rs

```diff
@@ -1,49 +1,79 @@
 use geom::{Distance, Polygon};
 
 use crate::{
-    svg, Color, Drawable, EdgeInsets, EventCtx, GeomBatch, GfxCtx, Key, Line, MultiKey, Outcome,
-    RewriteColor, ScreenDims, ScreenPt, ScreenRectangle, Text, Widget, WidgetImpl, WidgetOutput,
+    svg, text::Font, Color, ContentMode, ControlState, Drawable, EdgeInsets, EventCtx, GeomBatch,
+    GfxCtx, Key, Line, MultiKey, Outcome, RewriteColor, ScreenDims, ScreenPt, ScreenRectangle,
+    Text, Widget, WidgetImpl, WidgetOutput,
 };
 
 pub struct Button {
     /// When a button is clicked, `Outcome::Clicked` with this string is produced.
     pub action: String,
 
-    // Both of these must have the same dimensions and are oriented with their top-left corner at
+    // These must have the same dimensions and are oriented with their top-left corner at
     // 0, 0. Transformation happens later.
     draw_normal: Drawable,
     draw_hovered: Drawable,
+    draw_disabled: Drawable,
```

Comment:
Very in favor of having disabled buttons still be `Button`s. The `inactive()` turning it into a `JustDraw` was always just a lazy hack.